index.ts

```diff
@@ -35,6 +36,65 @@ export type WordPressArticle = {
   };
 };
 
+export type ImageSource = {
+  uri: string;
+  width: number;
+  height: number;
+};
+
+export type ImageData = {
+  id: number;
+  sources: ImageSource[];
+  caption: string;
+  alt: string;
+  position: string; // TODO: from mobile app: clarify possible string values
+};
+
+export type AudioData = { url: string; caption: string };
+
+export type VideoData = {
+  url: string;
+  caption: string;
+  aspectRatio: number;
+};
+
+export type Block = Omit<ParsedBlock, "innerBlocks"> & {
+  innerBlocks: Block[];
+  innerContent: string[];
```

Comment:
Technically, we probably don't need to specify `innerContent` because `innerContent` is already specified in the `ParsedBlock` type.

<img width="580" alt="Screenshot 2024-04-13 at 5 20 13 PM" src="https://github.com/michigandaily/wputils/assets/44459660/3e1269f0-33cb-4f27-bad4-9faef7bb5779">

The `Omit<> & {}` construction here essentially says take all of the type properties from `ParsedBlock` (which comes from WordPress), remove the `innerBlocks` property, and then extend it with our own type properties. We are extending it because we want the indexing and block data types for `innerBlocks`. 

More about Omit here: <https://www.typescriptlang.org/docs/handbook/utility-types.html#omittype-keys>

Reply:
We do use `innerContent` to render some mobile blocks! Notably used for lists.

Reply:
oh that makes sense, so true